crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1894,125 +2020,102 @@ impl SymbolicationActor {
-                                DebugId::from_str(&module.debug_identifier()).unwrap_or_default(),
-                                object_info_from_minidump_module(object_type, module),
-                            )
-                        })
-                        .collect();
-
-                    // Finally iterate through the threads and build the stacktraces to
-                    // return, marking modules as used when they are referenced by a frame.
-                    let requesting_thread_index: Option<usize> =
-                        process_state.requesting_thread().try_into().ok();
-                    let threads = process_state.threads();
-                    let mut stacktraces = Vec::with_capacity(threads.len());
-                    for (index, thread) in threads.iter().enumerate() {
-                        let registers = match thread.frames().get(0) {
-                            Some(frame) => map_symbolic_registers(
-                                frame.registers(minidump_state.system_info.cpu_arch),
-                            ),
-                            None => Registers::new(),
-                        };
-
-                        // Trim infinite recursions explicitly because those do not
-                        // correlate to minidump size. Every other kind of bloated
-                        // input data we know is already trimmed/rejected by raw
-                        // byte size alone.
-                        let frame_count = thread.frames().len().min(20000);
-                        let mut frames = Vec::with_capacity(frame_count);
-                        for frame in thread.frames().iter().take(frame_count) {
-                            let return_address =
-                                frame.return_address(minidump_state.system_info.cpu_arch);
-
-                            frames.push(RawFrame {
-                                instruction_addr: HexValue(return_address),
-                                package: frame.module().map(CodeModule::code_file),
-                                trust: frame.trust(),
-                                ..RawFrame::default()
-                            });
-                        }
-
-                        stacktraces.push(RawStacktrace {
-                            is_requesting: requesting_thread_index.map(|r| r == index),
-                            thread_id: Some(thread.thread_id().into()),
-                            registers,
-                            frames,
-                        });
-                    }
-
-                    Ok(procspawn::serde::Json(StackWalkMinidumpResult {
-                        modules,
-                        missing_modules,
-                        stacktraces,
-                        minidump_state,
-                    }))
+                |(cfi_caches, minidump_path, spawn_time)| {
+                    stackwalk_with_breakpad(cfi_caches, minidump_path, spawn_time)
```

Comment:
Kind of curious what forces you to still have a closure instead of passing the fn directly?

Reply:
The fact that the function takes 3 arguments, whereas the argument of `procspawn` takes a single argument which is a triple. I hate that this distinction exists.

(Also, as @Swatinem pointed out, it's cleaner to move the serialization stuff out of `stackwalk_with_{breakpad,rust_minidump}`.)